front-end/src/components/tables/player-tier.tsx

```diff
@@ -0,0 +1,205 @@
+import { Box, Paper, Table, TableBody, TableCell, TableContainer, TableRow, Typography, useTheme } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { GameRoles } from "../../../../Common/Interface/General/gameEnums";
+import PlayerStatModel from "../../../../Common/models/playerstat.model";
+import { calculateKDA, riotTimestampToMinutes, roundTo, toSearchName } from "../../../../Common/utils";
+import { getFlattenedLeaderboard } from "../../api/leaderboards";
+import { DPMStatGenerator } from "../../common/stats-generators/DPMStatGenerator";
+import { KDAStatGenerator } from "../../common/stats-generators/KDAStatGenerator";
+import { HeadCell, SortOrder } from "../../common/types";
+import SortableTableHead from "./helper-components/sortable-head";
+
+interface PlayerTierTableProps {
+  seasonId: string;
+  roleId: GameRoles;
+};
+
+interface LeaderboardType {
+  rank: number;
+  playerName: string;
+  role: GameRoles;
+  kda: number;
+  dpm: number;
+  gpm: number;
+  kpp: number;
+  wr: number;
+  games: number;
+}
+
+const headCells: HeadCell<LeaderboardType>[] = [
+  {
+    id: 'rank',
+    align: 'left',
+    disablePadding: false,
+    label: 'Rank'
+  },
+  {
+    id: 'role',
+    align: 'left',
+    disablePadding: false,
+    label: 'Role'
+  },
+  {
+    id: 'playerName',
+    align: 'left',
+    disablePadding: false,
+    label: 'Name'
+  },
+  {
+    id: 'wr',
+    align: 'center',
+    disablePadding: false,
+    label: 'WR',
+  },
+  {
+    id: 'kda',
+    align: 'center',
+    disablePadding: false,
+    label: 'KDA'
+  },
+  {
+    id: 'dpm',
+    align: 'center',
+    disablePadding: false,
+    label: 'DPM'
+  },
+  {
+    id: 'gpm',
+    align: 'center',
+    disablePadding: false,
+    label: 'GPM'
+  },
+  {
+    id: 'kpp',
+    align: 'center',
+    disablePadding: false,
+    label: 'KP%'
+  },
+  {
+    id: 'games',
+    align: 'center',
+    disablePadding: false,
+    label: 'Games'
+  },
+]
+
+function MapStatsToLeaderboard(data: PlayerStatModel[]): LeaderboardType[] {
+  return data.map((stat, i) => ({
+    rank: +i+1,
+    playerName: stat.player.name,
+    wr: roundTo((stat.win / stat.games)*100),
+    role: GameRoles[stat.lobbyPosition as keyof typeof GameRoles],
+    kda: calculateKDA(stat),
+    dpm: roundTo(stat.totalDamageDealtToChampions / riotTimestampToMinutes(stat.gameLength)),
+    gpm: roundTo(stat.goldEarned / riotTimestampToMinutes(stat.gameLength)),
+    kpp: roundTo((stat.kills + stat.assists) / stat.totalKillsOfTeam),
```

Comment:
Should use the statgenerators to calculate these so you dont duplicate code.

Reply:
Yeah. Thanks.